Sources/Apple/AppleAuthenticator+Models.swift

```diff
@@ -23,10 +23,11 @@ public extension AppleAuthenticator {
     public let email: Email
     public let expiresAt: Date
 
-    public var name: String {
-      [nameComponents?.givenName, nameComponents?.familyName]
+    public var name: String? {
+      let fullName = [nameComponents?.givenName, nameComponents?.familyName]
         .compactMap { $0 }
         .joined(separator: " ")
+      return !fullName.isEmpty ? fullName : nil
```

Comment:
I don't really like this code because it creates an unnecessary intermediate array (heap allocations and requires two iterations over the array to compute the value.

Here is an alternative:

```swift
  guard let givenName = nameComponents?.givenName else {
    return nameComponents?.familyName
  }
  guard let familyName = nameComponents?.familyName else {
    return givenName
  }
  return "\(givenName) \(familyName)"
```

Reply:
Good point. I like it 👌

Reply:
https://github.com/poviolabs/PovioKitAuth/pull/27/commits/ef38881431b8edcbd76b4cb4fda1edd76ccde89f